pkg/apis/pipeline/v1alpha1/artifact_bucket.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
```

Comment:
what do you think about putting the implementations of the artifacts interface into `package artifacts` ?

Reply:
I originally had the code in the `artifacts` package, but had to move it to be able to use existing variables (such as gsutilImage) which are defined in the api package. I would get a circular dependency.  This will probably need a bigger refactor which I didn't want to do here since this PR is big already